examples/java-matter-controller/java/src/com/matter/controller/commands/common/FutureResult.java

```diff
@@ -0,0 +1,62 @@
+/*
+ *   Copyright (c) 2023 Project CHIP Authors
+ *   All rights reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ *
+ */
+
+package com.matter.controller.commands.common;
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class FutureResult implements Result {
+  private RealResult realResult;
+  private boolean isReady = false;
```

Comment:
do we need both isReady and realResult? Could realResult be Optional or nullable and a missing/null represent not ready? Would also prefer optional instead of nullable.

Reply:
no need for isReady.